tests/CMakeLists.txt

```diff
@@ -45,7 +45,7 @@ function(ConfigureTestInternal TEST_NAME TEST_SRC)
   add_executable(${TEST_NAME} "${TEST_SRC}")
   target_include_directories(${TEST_NAME} PRIVATE "$<BUILD_INTERFACE:${RMM_SOURCE_DIR}>")
   target_link_libraries(${TEST_NAME} GTest::gmock GTest::gtest GTest::gmock_main GTest::gtest_main
-                        pthread rmm)
+                        pthread rmm dl)
```

Comment:
Should this be added to the main `rmm:rmm` target as opposed to the tests?

Reply:
@kkraus14 I'm not exactly sure what forced this change but it seems likely to me that other consumers of the `rmm::rmm` target will need this library as well. Adding it to the main target with `target_link_libraries(rmm INTERFACE dl)` seems like a good move.

I was just more hesitant to change the main target over the testing target. I'll make the change.